src/observers/OmnisharpStatusBarObserver.ts

```diff
@@ -0,0 +1,191 @@
+                this.render();
+                break;
+            case ActiveTextEditorChanged.name:
+                this.render();
+                break;
+            case OmnisharpServerOnStop.name:
+                this.handleOmnisharpServerOnStop();
+                break;
+            case OmnisharpServerOnStart.name:
+                this.handleOmnisharpServerOnStart(<OmnisharpServerOnStart>event);
+                break;
+        }
+    }
+
+    private updateProjectInfo = (server: OmniSharpServer) => {
+        this.firstUpdateProject = false;
+        serverUtils.requestWorkspaceInformation(server).then(info => {
+
+            interface Project {
+                Path: string;
+                SourceFiles: string[];
+            }
+
+            let fileNames: DocumentFilter[] = [];
+            let label: string;
+
+            function addProjectFileNames(project: Project) {
+                fileNames.push({ pattern: project.Path });
+
+                if (project.SourceFiles) {
+                    for (let sourceFile of project.SourceFiles) {
+                        fileNames.push({ pattern: sourceFile });
+                    }
+                }
+            }
+
+            function addDnxOrDotNetProjects(projects: Project[]) {
+                let count = 0;
+
+                for (let project of projects) {
+                    count += 1;
+                    addProjectFileNames(project);
+                }
+
+                if (!label) {
+                    if (count === 1) {
+                        label = basename(projects[0].Path); //workspace.getRelativePath(info.Dnx.Projects[0].Path);
+                    }
+                    else {
+                        label = `${count} projects`;
+                    }
+                }
+            }
+
+            // show sln-file if applicable
+            if (info.MsBuild && info.MsBuild.SolutionPath) {
+                label = basename(info.MsBuild.SolutionPath); //workspace.getRelativePath(info.MsBuild.SolutionPath);
+                fileNames.push({ pattern: info.MsBuild.SolutionPath });
+
+                for (let project of info.MsBuild.Projects) {
+                    addProjectFileNames(project);
+                }
+            }
+
+            // show .NET Core projects if applicable
+            if (info.DotNet) {
+                addDnxOrDotNetProjects(info.DotNet.Projects);
+            }
+
+            // set project info
+            this.projectStatus = new Status(fileNames);
+            SetStatus(this.projectStatus, '$(flame) ' + label, 'o.pickProjectAndStart');
+            SetStatus(this.defaultStatus, '$(flame) Switch projects', 'o.pickProjectAndStart');
+            this.render();
+        });
+    }
+
+    private render = () => {
+        let activeTextEditor = this.vscode.window.activeTextEditor;
+        if (!activeTextEditor) {
+            this.statusBar.hide();
```

Comment:
FWIW, I've wondered at times whether we should tweak this bit of code. Essentially, this means that the status bar item is not visible unless a file is open in the editor, which is a little weird IMO. What do you think?